### PR TITLE
ci: add workflows

### LINKED ISSUE #10: CI/CD plans
We'd like to get CI/CD set up quickly for this project. @sergioramos notes that for MacOS testing we'll need to sign our code, as noted in the Tauri docs: https://tauri.app/v1/guides/distribution/sign-macos.

@jonaskruckenberg @denjell-crabnebula any other tips for us on CI/CD for Tauri apps? Obviously there's the testing section in your docs but if there are any practical lessons you've learned that haven't made it into the docs yet we'd love to hear them! In particular we're concerned about testing on Windows since we don't have any Windows users. Any pro tips on how to build apps to make them maximally compatible across Mac and Windows?

### PR DESCRIPTION
**Pull request by sehyod**
Fixes #10

### CODE REVIEW

File: .github/workflows/on-push.yml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  pytest:
+    needs: setup-python
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Python
+        uses: ./.github/actions/setup-python
+      - name: PyTest
+        run: poetry run pytest
```
Comment (gjreda) [inline .github/workflows/on-push.yml:85]: @sehyod @sergioramos Let's change this line to look only in the test directory for now. It shouldn't matter here, but if you run this line as is in our repo, pytest picks up a lot of other things that aren't truly tests.

```suggestion
        run: poetry run pytest python/tests
```

I tried to push a change up to test this, but I don't have `workflow` scope on the repo.

```
 ! [remote rejected] initial-ci-cd -> initial-ci-cd (refusing to allow a Personal Access Token to create or update workflow `.github/workflows/on-push.yml` without `workflow` scope)
error: failed to push some refs to 'https://github.com/refstudio/refstudio'
```

File: .github/actions/setup-python/action.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    with:
+      python-version: "3.11"
+    id: setup-python
+  - uses: snok/install-poetry@v1
+    with:
+      virtualenvs-create: true
+      virtualenvs-in-project: true
+      installer-parallel: true
+  - id: cached-poetry-dependencies
+    uses: actions/cache@v3
```
Comment (gjreda) [inline .github/actions/setup-python/action.yml:16]: I would try removing the cache until we get a full workflow running properly, and then we can add it back in.

**Comment by gjreda** [comment]
@sehyod @sergioramos Looking at [the logs from this workflow](https://github.com/refstudio/refstudio/actions/runs/5145249529/jobs/9262639073#step:3:118), it does seem like the Python dependencies are being installed via poetry. Is there a way to verify?

I'd expect there to be some install dialog in the section below. Maybe something with the dependency caching we are using?

image

**Comment by sehyod** [comment]
@gjreda Thank you for looking into this issue! I have pushed the changes you suggested and added the verbosity option to `poetry install` to see what is going on. From the [new logs](https://github.com/refstudio/refstudio/actions/runs/5153570286/jobs/9280984530), I think dependencies are indeed installed by poetry

**Pull request merged**